apps/checkout-app/frontend/utils.ts

```diff
@@ -43,13 +45,46 @@ export const unflattenSettings = <T, S extends Node>(
   return unflattenedSettings;
 };
 
+/**
+ * Merges settings. To be used when default and saved settings may differ (e.g. after app update).
+ * @param defaultSettings
+ * @param savedSettings
+ * @returns Returns either previously saved settings values or default settings values. If settings values are present in default and saved at the same time, then saved value is returned.
+ */
+export const mergeSettingsValues = (
+  defaultSettings: UnknownSettingsValues,
+  savedSettings: UnknownSettingsValues
+) => {
+  const settings = {
+    ...defaultSettings,
+    ...savedSettings,
+  };
+
+  const mergedSettings = Object.keys(settings).reduce((mergedSettings, key) => {
```

Comment:
So why not return reduce right away like in the proposed change?

Reply:
Sure, fixed.